greatarc/location.py

```diff
@@ -9,7 +9,7 @@ def __init__(self, name, latitude, longitude):
     def get_coordinates(self):
         return f"Latitude: {self.latitude}, Longitude: {self.longitude}"
 
-    def calculate_distance(self, other_location):
+def calculate_distance(lat1, lon1, lat2, lon2, unit='km'):
```

Comment:
There are two issues: 1) you made new references to lat1/lon1, but did not use these new arguments in the following calculation. 2) the unit parameter is defined as default properly but is not effective in the actual calculation/conversion.